src/components/DialogAction/index.jsx

```diff
@@ -63,7 +69,9 @@ function DialogAction(props) {
       onComplete(result);
     }
 
-    setActionExecuting(false);
+    if (!cancelled) {
+      setActionExecuting(false);
+    }
```

Comment:
This is exactly what I had in mind :) We should probably also wrap the other `setActionExecuting` with the if statement.

Reply:
Good point. We don't have any onError handlers that would unmount at the moment, but we certainly could in the future...